extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTableAdapter.java

```diff
@@ -186,17 +204,27 @@ public synchronized Schema currentSchema() {
         return table.schema();
     }
 
+    /**
+     * Retrieve the current {@link Schema schema} of an Iceberg table.
+     */
+    public synchronized Map<Integer, Schema> schemas() {
+        refresh();
+        return table.schemas();
```

Comment:
We could consider a defensive copy here. Implementations we've looked at return immutable maps.